luminaire/model/window_density.py

```diff
@@ -545,23 +549,47 @@ def _get_result(self, input_df=None, detrend_order=None, agg_data_model=None, va
             execution_data_avg = np.mean(execution_data)
             # If detrending is needed, we scale the scoring data accordingly using the agg_dat_model forecast
             if detrend_order > 0:
+                snapshot_len_max = min(len(agg_data), len_req_agg_data_model)
+                agg_data_trunc = np.array(agg_data)[:, 1][-snapshot_len_max:]
+                data_adjust_forecast = []
                 try:
-                    data_adjust_forecast = agg_data_model.score(execution_data_avg, scoring_datetime)['Prediction'] \
-                        if agg_data_model else agg_data[-1][-1]
+                    if agg_data_model and len(agg_data) > len_req_agg_data_model:
+                        score = agg_data_model.score(execution_data_avg, scoring_datetime)
+                        data_adjust_forecast.append(score['Prediction'])
+                        data_adjust_forecast.append(score['CILower'])
+                        data_adjust_forecast.append(score['CIUpper'])
```

Comment:
So, you are directly appending all values to single list. Can you explain why? Since they are different like Prediction, CILower and CIUpper. Can we add some comments for this.

Reply:
I'll add a comment. This makes sure the adjustment of the original data is made keeping the prediction uncertainty in mind.